homeassistant/components/insteon/api/config.py

```diff
@@ -0,0 +1,277 @@
+
+
+def add_x10_device(hass: HomeAssistant, x10_device: dict):
+    """Add an X10 device to the Insteon integration."""
+    config_entry = get_insteon_config_entry(hass)
+    options_config = {**config_entry.options}
+    x10_config = config_entry.options.get(CONF_X10, [])
+
+    # Check that we are not adding an existing device
+    for curr_device in x10_config:
+        if (
+            curr_device[CONF_HOUSECODE] == x10_device[CONF_HOUSECODE]
+            and curr_device[CONF_UNITCODE] == x10_device[CONF_UNITCODE]
+        ):
+            raise ValueError("Duplicate X10 device")
+
+    x10_config.append(x10_device)
+    options_config[CONF_X10] = x10_config
+    hass.config_entries.async_update_entry(entry=config_entry, options=options_config)
+    async_dispatcher_send(hass, SIGNAL_ADD_X10_DEVICE, x10_device)
+
+
+def remove_x10_device(hass: HomeAssistant, housecode: str, unitcode: int):
+    """Remove an X10 device from the config."""
+
+    config_entry = get_insteon_config_entry(hass)
+    new_options = {**config_entry.options}
+    try:
+        new_options.pop(CONF_X10)
+    except KeyError:
+        # There are no X10 devices so nothing to do
+        return
+
+    new_x10 = [
+        existing_device
+        for existing_device in config_entry.options[CONF_X10]
+        if existing_device[CONF_HOUSECODE].lower() != housecode.lower()
+        or existing_device[CONF_UNITCODE] != unitcode
+    ]
+
+    if new_x10:
+        new_options[CONF_X10] = new_x10
+    hass.config_entries.async_update_entry(entry=config_entry, options=new_options)
+
+
+def add_device_overide(hass: HomeAssistant, override: dict[str, str]):
+    """Add an Insteon device override."""
+
+    if has_device_override(hass, Address(override[CONF_ADDRESS])):
+        raise ValueError("Duplicate override")
+
+    config_entry = get_insteon_config_entry(hass)
+    options_config = {**config_entry.options}
+    override_config = config_entry.options.get(CONF_OVERRIDE, [])
+    override_config.append(override)
+    options_config[CONF_OVERRIDE] = override_config
+    hass.config_entries.async_update_entry(entry=config_entry, options=options_config)
+    async_dispatcher_send(hass, SIGNAL_ADD_DEVICE_OVERRIDE, override)
+
+
+def has_device_override(hass: HomeAssistant, address: Address):
+    """Test if a device has a device override."""
+
+    config_entry = get_insteon_config_entry(hass)
+    override_config = config_entry.options.get(CONF_OVERRIDE, [])
+    for existing_override in override_config:
+        if Address(existing_override[CONF_ADDRESS]) == address:
+            return True
+    return False
+
+
+def remove_device_override(hass: HomeAssistant, address: Address):
+    """Remove a device override from config."""
+
+    config_entry = get_insteon_config_entry(hass)
+    new_options = {**config_entry.options}
+    try:
+        new_options.pop(CONF_OVERRIDE)
+    except KeyError:
+        # There are no overrides so nothing to do
+        return
```

Comment:
Same comment as before, why do we have a special case for this?

Reply:
Yeah, I see what you are saying. This, combined with the `if new_overrides:` below was intended to avoid an empty list but that is really not helpful.